modules/openid-federation-common/src/commonTest/kotlin/com/sphereon/oid/fed/common/mime/JsonUrlEncoderTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.sphereon.oid.fed.common.mime
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+
+@Serializable
+data class PreAuthorizedCode(val pre_authorized_code: String)
+
+@Serializable
+data class Grants(val urn_ietf_params_oauth_grant_type_pre_authorized_code: PreAuthorizedCode)
+
+@Serializable
+data class TestData(
+    val grants: Grants,
+    val credential_configuration_ids: List<String>,
+    val credential_issuer: String
+)
+
+class JsonUrlEncoderTest {
+    private val originalJson =
+        """{"grants":{"urn:ietf:params:oauth:grant-type:pre-authorized_code":{"pre-authorized_code":"a"}},"credential_configuration_ids":["Woonplaatsverklaring"],"credential_issuer":"https://agent.issuer.bd.demo.sphereon.com"}"""
```

Comment:
We should always use dummy data when creating tests. Please remove all references to other projects.

Reply:
Fixed in next commit